src/sentry/api/event_search.py

```diff
@@ -304,13 +307,10 @@ def _flatten(seq):
             else:
                 yield item
 
-    if not (children and isinstance(children, list) and isinstance(children[0], list)):
-        return children
-
-    children = [child for group in children for child in _flatten(group)]
-    children = [_f for _f in _flatten(children) if _f]
+    if not isinstance(children, list):
+        return [children]
```

Comment:
this is a ~slight behaviour change as well  -- the old code would sometimes be passed a raw `Node`.  and the parser "got lucky" because `Node.__iter__` iterates its children which was always `[]` in those cases